sorc/chgres_cube.fd/wam_climo_data.f90

```diff
@@ -0,0 +1,2796 @@
+!! @param[in] meter logical true or false
+!! @author Hann-Ming Henry Juang
+      subroutine meters(meter) 
+      use wam_gtd7bk_mod, only: imr
+!
+      logical meter 
+      save 
+      imr=0 
+      if(meter) imr=1 
+      end subroutine meters                                         
+
+!-----------------------------------------------------------------------
+!> Calculate scale height (km)                                      
+!! @param[in] alt  altitude [km]
+!! @param[in] xm   molecular weihjt
+!! @param[in] temp temperature
+!! @return scalh   scale height
+!!
+!! @author Hann-Ming Henry Juang 
+      function scalh(alt,xm,temp) 
+      use gettemp_mod, only: gsurf,re
+!
+      save 
+      data rgas/831.4/ 
+      g=gsurf/(1.+alt/re)**2 
+      scalh=rgas*temp/(g*xm) 
+      return 
+      end function scalh                            
+
+!-----------------------------------------------------------------------
+!> Calculate g(l) function for upper thermosphere parameters                                   
+!! @param[in] yrd    year and day as yyddd                                    
+!! @param[in] sec    ut(sec)                                                  
+!! @param[in] lat    geodetic latitude(deg)                                  
+!! @param[in] long   geodetic longitude(deg)                                
+!! @param[in] tloc   local apparent solar time(hrs)                           
+!! @param[in] f107a  3 month average of f10.7 flux                          
+!! @param[in] f107   daily f10.7 flux for previous day                       
+!! @param[in] ap     magnetic index(daily) or when sw(9)=-1. :                 
+!!                array containing:                                         
+!!                ap(1) daily ap                                              
+!!                ap(2) 3 hr ap index for current time                        
+!!                ap(3) 3 hr ap index for 3 hrs before current time           
+!!                ap(4) 3 hr ap index for 6 hrs before current time           
+!!                ap(5) 3 hr ap index for 9 hrs before current time           
+!!                ap(6) average of eight 3 hr ap indicies from 12 to 33 hrs pr
+!!                         to current time                                    
+!!                ap(7) average of eight 3 hr ap indicies from 36 to 59 hrs pr
+!!                         to current time                                    
+!! @param[in] p  pressure level(mb)                                     
+!! @return globe7  version of global
+!!
+!! @author Hann-Ming Henry Juang
+      function globe7(yrd,sec,lat,long,tloc,f107a,f107,ap,p) 
+      use gettemp_mod, only: tinf=>tinfg,t=>tt,    &
+                             sw,swc,isw,        &
+                             plg,ctloc,stloc,c2tloc,s2tloc,c3tloc,s3tloc,    &
+                             day,df,dfa,apd,apdf,apt,xlong,iyr
+      real lat, long 
+      dimension p(*),sv(25),ap(*) 
```

Comment:
I see some arguments are dimensioned using "*". Will modern compilers balk at that?